packages/fe-tw/src/components/Admin/auditManagement/index.tsx

```diff
@@ -0,0 +1,292 @@
+"use client";
+
+import { Form, Formik, FormikProps } from "formik";
+import { Card, CardContent, CardHeader } from "@/components/ui/card";
+import React, { useEffect, useState, useRef } from "react";
+import { toast } from "sonner";
+import { Delete } from "lucide-react";
+import { Label } from "@/components/ui/label";
+import { Input } from "@/components/ui/input";
+import {
+   Select,
+   SelectContent,
+   SelectItem,
+   SelectTrigger,
+   SelectValue,
+} from "@/components/ui/select";
+import { useBuildings } from "@/hooks/useBuildings";
+import { pinata } from "@/utils/pinata";
+import { Textarea } from "@/components/ui/textarea";
+import { Button } from "@/components/ui/button";
+import { useBuildingAudit } from "@/hooks/useBuildingAudit";
+import { tryCatch } from "@/services/tryCatch";
+import { TxResultToastView } from "@/components/CommonViews/TxResultView";
+import { AuditFormValues, initialValues, validationSchema } from "./helpers";
+import { LoadingView } from "@/components/LoadingView/LoadingView";
+
+export function AuditManagementForm() {
+   const formikRef = useRef<FormikProps<AuditFormValues>>(undefined);
+   const { buildings } = useBuildings();
+   const [selectedBuildingAddress, setSelectedBuildingAddress] = useState<`0x${string}`>();
+   const { addAuditRecordMutation, updateAuditRecordMutation, revokeAuditRecord, auditDataLoading, auditData } = useBuildingAudit(selectedBuildingAddress!);
+
+   useEffect(() => {
+      if (!!auditData?.data) {
+         Object.keys(auditData.data).forEach((key) => {
+            formikRef.current?.setFieldValue(key, (auditData.data as any)[key]);
+         });
+      } else {
+         formikRef.current?.setFormikState((prev) => ({
+            ...prev,
+            values: initialValues,
+         }));
+      }
+   }, [auditData]);
+
+   const handleSubmit = async (values: AuditFormValues, resetForm: () => void) => {
+      try {
+         const fileName = `audit-${selectedBuildingAddress}-${Date.now()}`;
+         const keyRequest = await fetch("/api/pinataKey");
+         const keyData = await keyRequest.json();
+
+         const { IpfsHash: newIpfsHash } = await pinata.upload
+            .json(values, {
+               metadata: { name: fileName },
+            })
+            .key(keyData.JWT);
+
+         if (!newIpfsHash) {
+            toast.error(
+               <TxResultToastView
+                  title="Error during uploading Audit data to IPFS"
+                  txError="Error during uploading Audit data to IPFS"
```

Comment:
Since it is not on-chain operation, so there is not transaction, so no need to use `TxResultToastView`, just plaing `toast.error`